services/ad-server/src/public/js/ad-server-lib.js

```diff
@@ -162,20 +181,27 @@ class AdServerLib {
 
   addContainerFrame({divId, size, type, isFencedFrame}) {
     const containerFrameEl = document.createElement('iframe');
-    containerFrameEl.src = `${this.adServerOrigin}/ad-frame.html`;
-
-    const [width, height] = size;
-    containerFrameEl.width = width;
-    containerFrameEl.height = height;
+    const topLevelOrigin = encodeURI(window.location.origin);
+    containerFrameEl.src = `${this.adServerOrigin}/ad-frame.html?top-level-origin=${topLevelOrigin}`;
 
+    // Add a label
     const paragraphEl = document.createElement('p');
     paragraphEl.innerText = `${type} ad in ${
       isFencedFrame ? 'fenced frame' : 'iframe'
     }`.toUpperCase();
     paragraphEl.className = 'font-mono text-sm';
+    document.getElementById(divId).appendChild(paragraphEl);
+
+    if (type === 'image') {
+      const [width, height] = size;
+      containerFrameEl.width = width;
+      containerFrameEl.height = height;
+    } else if (type === 'video') {
+      containerFrameEl.width = 0;
```

Comment:
For an image ad, the image is actually rendered inside the container el iframe. 

For a video ad, what is rendered inside the container iframe is the HTML document that only contains the VAST URI.  That VAST URI is post-messaged out of the container iframe to the video player, and the ad is rendered outside of this container.  

Therefore, we hide this player since we have nothing to render in this frame, and only need to load the code that post-messages the VAST.